nav2_docking/opennav_docking/src/docking_server.cpp

```diff
@@ -164,7 +164,10 @@ DockingServer::on_cleanup(const rclcpp_lifecycle::State & /*state*/)
   tf2_buffer_.reset();
   docking_action_server_.reset();
   undocking_action_server_.reset();
-  dock_db_.reset();
+  if (dock_db_) {
```

Comment:
Cleanup is the opposite of configure. This would never be called if configure isn't called, so we know `dock_db_` must exist at this point

Reply:
This is already handled 

```
  if (!dock_db_->initialize(node, tf2_buffer_)) {
    on_cleanup(state);
    return nav2_util::CallbackReturn::FAILURE;
  }
```

Anything that can throw is already protected